runtime/src/accounts.rs

```diff
@@ -114,7 +114,10 @@ impl Accounts {
                 .filter(|key| !message.program_ids().contains(key))
             {
                 let (account, rent) = AccountsDB::load(storage, ancestors, accounts_index, key)
-                    .and_then(|(account, _)| rent_collector.update(account))
+                    .and_then(|(mut account, _)| {
+                        let rent_due = rent_collector.update(&mut account);
+                        Some((account, rent_due))
```

Comment:
I really want to see this on the store() side, because of what create_account() has to look like...

Reply:
that will be on separate PR. This PR will only have changes in rent_collector.

Reply:
hmm. @rob-solana So, design of deducting rent for credit-debit and readonly will be similar in nature. cool, I will think about this bit more, and get back.

Reply:
> hmm. On visualising, this design is less intuitive, instead we could do this:
> 
> 1. Define `ACCOUNT_STORAGE_OVERHEAD`, and assign it to some arbitrary value.
> 2. Modify calculation in a way that by using lamports per byte per year, we can calculate how much rent is due for an account for bytes of (`ACCOUNT_STORAGE_OVERHEAD` + data.len())

if you take approach #1, there are 2 variables that need to change in order to turn on rent...  approach #2 means you only have to change lamports_per_byte_year

Reply:
@rob-solana Those are part of same design. :) Yeah, now you can alter `lamports_per_byte_year` to turn on/off rent.  ACCOUNT_STORAGE_OVERHEAD is defined for sake of visibility in design.